FreeRTOS/Test/CMock/smp/multiple_priorities_no_timeslice/covg_multiple_priorities_no_timeslice_utest.c

```diff
@@ -627,4 +634,594 @@ void test_task_delay_until_with_config_assert( void )
+    uxSchedulerSuspended = pdTRUE;
+
+    vTaskGetInfo( &xTaskTCBs[0], &pxTaskStatus, xFreeStackSpace, taskState);
+    TEST_ASSERT_EQUAL((BaseType_t)1, pxTaskStatus.uxCurrentPriority);
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *                if( listLIST_ITEM_CONTAINER( &( pxTCB->xEventListItem ) ) != NULL )
+ *                {
+ *                   pxTaskStatus->eCurrentState = eBlocked;
+ *                }
+ *                ...
+ * @endcode
+ *
+ * Cover the case where the task is blocked.
+ */
+void test_coverage_vTaskGetInfo_blocked_task( void )
+{
+    TCB_t xTaskTCBs[ configNUMBER_OF_CORES + 1U ] = { NULL };
+    TaskStatus_t pxTaskStatus;
+    BaseType_t xFreeStackSpace = pdFALSE;
+    eTaskState taskState = eSuspended;
+
+    /* Setup the variables and structure. */
+    vListInitialise( &xSuspendedTaskList );
+    vListInitialise( &xPendingReadyList );
+
+    xTaskTCBs[ 0 ].uxPriority = 2;
+    xTaskTCBs[ 0 ].xTaskRunState = -1;
+    xYieldPendings[ 0 ] = pdFALSE;
+    pxCurrentTCBs[ 0 ] = &xTaskTCBs[ 0 ];
+    listINSERT_END( &xSuspendedTaskList, &xTaskTCBs[ 0 ].xStateListItem );
+
+    uxTopReadyPriority = 2;
+    uxSchedulerSuspended = pdTRUE;
+
+    xTaskTCBs[0].xEventListItem.pxContainer = (struct xLIST *) 1;
+    vTaskGetInfo( &xTaskTCBs[0], &pxTaskStatus, xFreeStackSpace, taskState);
+    TEST_ASSERT_EQUAL((BaseType_t)2, pxTaskStatus.uxCurrentPriority);
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *       if( taskTASK_IS_RUNNING( pxTCB ) == pdTRUE )
+ *            {
+ *                pxTaskStatus->eCurrentState = eRunning;
+ *            }
+ *            ...
+ * @endcode
+ *
+ * Cover the case in the taskTASK_IS_RUNNING() macro where the xTaskRunState
+ * is out of bounds.
+ */
+void test_coverage_vTaskGetInfo_oob_xTaskRunState( void )
+{
+    TCB_t xTaskTCBs[ configNUMBER_OF_CORES + 1U ] = { NULL };
+    TaskStatus_t pxTaskStatus;
+    BaseType_t xFreeStackSpace = pdFALSE;
+    eTaskState taskState = eSuspended;
+
+    xTaskTCBs[ 0 ].uxPriority = 1;
+    xTaskTCBs[ 0 ].xTaskRunState = configNUMBER_OF_CORES;
+    xYieldPendings[ 0 ] = pdFALSE;
+    pxCurrentTCBs[ 0 ] = &xTaskTCBs[ 0 ];
+
+    uxTopReadyPriority = 1;
+    uxSchedulerSuspended = pdTRUE;
+
+    vTaskGetInfo( &xTaskTCBs[0], &pxTaskStatus, xFreeStackSpace, taskState);
+    TEST_ASSERT_EQUAL((BaseType_t)1, pxTaskStatus.uxCurrentPriority);
+}
+
+/**
+ * @brief vTaskGetInfo - populate TaskStatus_t and eTaskState
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *       if( taskTASK_IS_RUNNING( pxTCB ) == pdTRUE )
+ *            {
+ *                pxTaskStatus->eCurrentState = eRunning;
+ *            }
+ *            ...
+ * @endcode
+ *
+ * Cover the case where xFreeStackSpace is pdFALSE, avoiding the free
+ * stack space query.
+ */
+
+static void test_prvInitialiseStack( TCB_t *pxTCB, const uint32_t ulStackDepth )
```

Comment:
For my understanding, this function is to setup the stack of a task to be returned by vTaskGetInfo.
The code looks like code snippet from tasks.c.
Can we try to make it cleaner only setup the following variables.
```
pxTCB->pxStack
pxTCB->EndOfStack
```
Also verify the value returned by vTaskGetInfo is as expected.